pkg/features/features.go

```diff
@@ -124,9 +126,37 @@ func (fg featureFlag) SetFeatureFlag(features []string) error {
 		fg[f] = isEnabled
 	}
 
+	// We make sure features are only turned on if their dependencies are met
+	// We memoize values to avoid computing the same dependency twice
+	memoizedValues := make(featureFlag)
+	for feature := range fg {
+		ValidateDependencies(feature, fg, memoizedValues)
```

Comment:
```suggestion
		_ = ValidateDependencies(feature, fg, memoizedValues)
```
to ignore the return value of the function.